### PR TITLE
GEOMAR Updates

### PR DESCRIPTION
**Pull request by seb-wahl**
I already merged the latest release (v6.17.2-32-gcd50c603) just now into geomar_dev (without any conflicts). So merging `geomar_dev` back into `release` should be straight forward. 
What's new:
- integration of NEMO 4.2 into ESM-Tools
- standalone NEMO simulations with NEMO4.2 using the ORCA05 and JRA55 v1.5-do forcing now technically works. No scientific validation as this is an intermediate version on our way towards the eORCA025 grid.
- NEMO, ECHAM6, OIFS postprocessing now works for sub-yearly and multiyear runs.
- improved file movements for FOCI (e.g. move output data instead of copying)

### CODE REVIEW

File: configs/components/oifs/oifs.yaml [hunk 1]
```diff
@@ -696,7 +704,7 @@ choose_with_xios:
                                         LXIOS: ".true."
                 outdata_files:
                         oifsnc: oifsnc
-                        atm: atm
+                        #atm: atm
```
Comment (mandresm) [inline configs/components/oifs/oifs.yaml]: I believe this is important for `AWICM3` and it shouldn't be removed. Is it giving any problem to `oifs` standalone or `focioifs`?

File: configs/components/oifs/oifs_grib_output_to_restart.sh [hunk 2]
```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+
```
Comment (mandresm) [inline configs/components/oifs/oifs_grib_output_to_restart.sh]: Please, indicate the author of the script and provide contact information. I don't have a problem with providing some shell scripts in the `configs`, as long as the configs don't get flooded with them, and as long as we don't have to support them

Comment (mandresm) [reply configs/components/oifs/oifs_grib_output_to_restart.sh]: Can someone also add a small comment for under which circumstances this script is needed?

File: configs/esm_software/esm_runscripts/defaults_geomar.yaml [hunk 3]
```diff
@@ -25,17 +25,36 @@ per_model_defaults:
       input:
          all_directions: link
       restart_in:
-         # seb-wahl: need to copy restart files by default
```
Comment (mandresm) [inline configs/esm_software/esm_runscripts/defaults_geomar.yaml:28]: Are you using this file at all? I thought you'd be doing the changes via the `foci` setup files. If that's the case you can remove this file. If you need this file around I have not problem with keeping it.

**Review by mandresm — CHANGES_REQUESTED** [review]
Hi @seb-wahl, thanks for your PR. I have some questions, but overall they are very minor and I'd expect a fast merge. Is there a particular script for standalone NEMO that you'd like to be added to `esm_tests`?

File: configs/esm_software/esm_runscripts/defaults_geomar.yaml (hunk 3 above)
Comment (seb-wahl) [reply configs/esm_software/esm_runscripts/defaults_geomar.yaml:28]: This file can be removed. It's from past times where file_movements within e.g. setup yaml files didn't work (properly). I'll remove it in my next commit.

File: configs/components/oifs/oifs_grib_output_to_restart.sh (hunk 2 above)
Comment (seb-wahl) [reply configs/components/oifs/oifs_grib_output_to_restart.sh]: @joakimkjellsson I will add some comments in the next commit. If it's more a personal script, we should move it somewhere else.

File: configs/components/oifs/oifs.yaml (hunk 1 above)
Comment (seb-wahl) [reply configs/components/oifs/oifs.yaml]: @joakimkjellsson can you comment on this. I added it again, did a short test, and everything seems to be fine. I'll add a commit where the `#`is removed.

**Review by mandresm — APPROVED** [review]
#approve-changes

**Comment by mandresm** [comment]
#approve-changes

**Comment by mandresm** [comment]
#bump

**Comment by mandresm** [comment]
#bump #minor

**Pull request merged**